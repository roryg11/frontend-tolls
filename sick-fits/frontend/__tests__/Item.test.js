import Item from "../components/Item";
import { shallow, enzyme } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
    id: "HELLO3451",
    title: "Sweet Kicks",
    price: "45059",
    image: "kicks.jpg",
    largeImage: "bigKicks.jpg"
};

describe('<Item/>', ()=>{
    it('renders', ()=>{
        shallow(<Item item={fakeItem}/>);
    });

    it('matches the snapshot', ()=>{
        const itemWrapper = shallow(<Item item={fakeItem}/>);
        expect(toJSON(itemWrapper)).toMatchSnapshot();
    });

    it('renders the image properly', ()=>{
        const itemWrapper = shallow(<Item item={fakeItem}/>);
        const img = itemWrapper.find('img');
        expect(img.props().src).toEqual(fakeItem.image);
        expect(img.props().alt).toEqual(fakeItem.title);
    });

    it('renders the price and title properly', ()=>{
        const itemWrapper = shallow(<Item item={fakeItem}/>);
        const PriceTag = itemWrapper.find('PriceTag');
        const title = itemWrapper.find('Title a');
        expect(title.children().text()).toEqual(fakeItem.title);
        expect(PriceTag.children().text()).toBe("$450.59");
    });

    it('renders out the buttons for the item', ()=>{
        const itemWrapper = shallow(<Item item={fakeItem}/>);
        const buttonList = itemWrapper.find('.buttonList');
        expect(buttonList.children()).toHaveLength(3);
        expect(buttonList.find('Link')).toHaveLength(1);
        expect(buttonList.find('AddToCart')).toHaveLength(1);
        expect(buttonList.find('DeleteItem')).toHaveLength(1);
    });
});