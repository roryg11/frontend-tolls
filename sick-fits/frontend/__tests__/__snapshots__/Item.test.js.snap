// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Sweet Kicks"
    src="kicks.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "HELLO3451",
          },
        }
      }
    >
      <a>
        Sweet Kicks
      </a>
    </Link>
  </Title>
  <PriceTag>
    $450.59
  </PriceTag>
  <p />
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "HELLO3451",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="HELLO3451"
    />
    <DeleteItem
      id="HELLO3451"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
